### PR TITLE
Define WeakSet in lib/core.js

### LINKED ISSUE #832: WeakSet is not defined in lib/core.js.
However, `WeakMap` is...

### PR DESCRIPTION
**Pull request by paulcapron**
Closes #832.

I’m sorry I didn’t add tests, but I didn’t know where to put them. There are some tests for `Set` in `tests/iterable`, but a `WeakSet` is *not* enumerable, so this wasn’t a right place. And there are no test for `WeakMap`. Maybe `tests/weakstructs` should be created?

### CODE REVIEW

File: lib/core.js [hunk 1]
```diff
@@ -386,6 +386,12 @@ declare class Set<T> {
     values(): Iterator<T>;
 }
 
+declare class WeakSet<T: Object> {
```
Comment (mroch) [inline lib/core.js]: i think you need to add a constructor, like

```
<V>(iterable: ?Iterable<V>): WeakSet<V>
```

not sure how that works when the param is missing... hopefully we leave it unresolved and the first call to `add` will lock it down. cc @avikchaudhuri

**Comment by samwgoldman** [comment]
I noticed the missing constructor, too. I looked around and realized that `Set` and `WeakMap` are also missing constructors.

**Comment by mroch** [comment]
@facebook-github-bot import

**Comment by mroch** [comment]
no worries, I added the constructor(s) and some tests. thanks @paulcapron

**Pull request closed without merging**